src/shogun/multiclass/KNN.cpp

```diff
@@ -161,9 +162,11 @@ CMulticlassLabels* CKNN::apply_multiclass(CFeatures* data)
 	if (m_k == 1)
 		return classify_NN();
 
-	ASSERT(m_num_classes>0)
-	ASSERT(distance)
-	ASSERT(distance->get_num_vec_rhs())
+	REQUIRE(
+	    m_num_classes > 0,
+	    "Number of classes is zero. You may need to call train()\n");
```

Comment:
"Number of classes is zero" is not helpful here.
I would rather say "Machine is not trained"